### ISSUE TITLE
non-empty request body was deprecated since Docker v1.10 and removed in v1.12

### ISSUE THREAD

**Issue by shaunmulligan** [resin-io/resin-cli#534]
`resin local push` breaks with resinOS devices running docker17.03 because part of the docker API was removed in docker 1.12
```
rdt push failed. Error: Error while starting container test: Error: (HTTP code 400) unexpected - starting container with non-empty request body was deprecated since v1.10 and removed in v1.12  Error: Error while starting container test: Error: (HTTP code 400) unexpected - starting container with non-empty request body was deprecated since v1.10 and removed in v1.12
```

**Comment by Arwid** [comment]
+1

**Comment by pimterry** [comment]
@lekkas this is assigned to you - are you looking into it? Any progress?

**Comment by imrehg** [comment]
Ping, as more and more devices our out there with the new docker.

**Comment by lekkas** [comment]
@imrehg @pimterry sorry I completely missed the assignment & first ping. I'm not looking into this (disclaimer: in general I try to steer clear of the CLI during the last couple of months) but I did a quick research now:

The fix should be relatively easy by removing the passed `HostConfig` options from the body of the `startContainer()` request (https://github.com/resin-io-modules/resin-sync/blob/master/lib/docker-utils.coffee#L211) and adding them in `createContainer()` . 

The most important thing here is to make sure that we keep backwards compatibility. @imrehg @shaunmulligan do we have a range of Docker versions running on devices in the field where `resin local push` can be used? We can then use https://docs.docker.com/engine/api/v1.30/#section/Versioning to make the proper API call adjustments.

**Comment by shaunmulligan** [comment]
@lekkas I think we basically have 3 versions of docker. 
* 1.4 (which doesn't support local mode anyways)
* 1.10.3
* 17.03
* 17.06 (coming soon) 
So I think we really just need to make sure we can interact with the last three

**Comment by imrehg** [comment]
In the forums too: https://forums.resin.io/t/docker-container-not-building-in-resinio-image/1532/9

**Comment by shaunmulligan** [comment]
@emirotin can you look into this issue?

**Comment by emirotin** [comment]
Thanks for the input, will try to look into it asap.

**Comment by emirotin** [comment]
@shaunmulligan @imrehg I've pushed the branch `132-move-host-config-opts` that should include the fix
If you have the CLI repo locally you need to do this to verify it (if you have time to help me with testing of course):

```
git pull
git checkout 132-move-host-config-opts
npm update
npm run build
```

Then use the `./bin/resin`, it'll pick your existing auth

**Comment by shaunmulligan** [comment]
@emirotin I tried to follow the instruction there, but I can't find any branch by that name? Do I need to update to that branch in the resin-sync module?

**Comment by shaunmulligan** [comment]
@emirotin just tested this on the latest v2.2.0 resinOS and it works great!

**Comment by emirotin** [comment]
Ah, I've put the branch name for the sync, sorry :(

**Comment by emirotin** [comment]
Published as 6.3.0
